bundles/framework/rpc/service/RpcService.js

```diff
@@ -0,0 +1,140 @@
+import { arrayToObject, domainMatch } from '../util/RPCUtil';
+
+/**
+ * @class Oskari.mapframework.bundle.rpc.service.RpcService
+ * Allows extend RPC supported funtions
+ *
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.rpc.service.RpcService',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     *
+     */
+
+    function (instance) {
+        this.instance = instance;
+        this.conf = this.instance.conf || {};
+        this.channel = this.instance._channel;
+        this._log = Oskari.log('RpcService');
+        this._availableFunctions = {};
+        this._allowedFunctions = this.conf.allowedfunctions || [];
+        this._allowedFunctionsFromConf = Array.isArray(this.conf.allowedfunctions);
+    }, {
+        __name: 'Rpc.RpcService',
+        __qname: 'Oskari.mapframework.bundle.rpc.service.RpcService',
+
+        getQName: function () {
+            return this.__qname;
+        },
+
+        getName: function () {
+            return this.__name;
+        },
+
+        /**
+         * @method init
+         * Initializes the service
+         */
+        init: function () {
+        },
+
+        /**
+         * @method addFunction
+         * Add function to RPC
+         *
+         * @param {String} name function name
+         * @param {Function} func function that should be added
+         */
+        addFunction: function (name, func) {
+            if (!func && typeof func !== 'function') {
+                return;
+            }
```

Comment:
We still need to check that the function has a name. I think it's best to throw an error to make it more noticeable.
```suggestion
            if (!name || typeof func !== 'function') {
                throw new Error('Tried registering function without name or implementation!');
            }
```